video/out/mac/common.swift

```diff
@@ -395,7 +406,58 @@ class Common: NSObject {
     }
 
     func updateICCProfile() {
-        log.sendWarning("updateICCProfile not implemented")
+        guard let windowColorSpace = window?.screen?.colorSpace else {
+            log.sendWarning("Couldn't update ICC Profile, no color space available")
+            return 
+        }
+
+        colorSpace = windowColorSpace
+
+        if #available(macOS 10.11, *) {
+
+            // Experimental #HDR on #macOS        
+            var name = "" as CFString
+
+// HDR is supported fully only on macOS 10.15 and higher
+// some of the HDR color spaces are available on 10.14 though
+#if HAVE_MACOS_10_15_FEATURES
+                switch (mpv?.macOpts.macos_hdr_csp)
+                {
+                    case 0: if #available(macOS 10.14.6, *) {
+                                name = CGColorSpace.displayP3_HLG
+                            }   
+                            break;
+                    case 1: if #available(macOS 10.14.6, *) {
+                                name = CGColorSpace.displayP3_PQ_EOTF
+                            }   
+                            break;
+                    case 2: if #available(macOS 10.14.3, *) {
+                                name = CGColorSpace.extendedLinearDisplayP3
+                            }
+                            break;
+                    case 3: name = CGColorSpace.itur_2020
+                            break;
+                    case 4: if #available(macOS 10.15.6, *) {
+                                name = CGColorSpace.itur_2020_HLG 
+                            }
+                            break;
+                    case 5: if #available(macOS 10.14.6, *) {
+                                name = CGColorSpace.itur_2020_PQ_EOTF
+                            }
+                            break;
+                    default: 
+                            break;
+                }
+#endif            
+            // Experimental #HDR on #macOS        
+            if (name as String != "")
+            {
+                (view?.layer as! CAOpenGLLayer).colorspace = CGColorSpace(name: name)
+            } else
+            {
+                (view?.layer as! CAOpenGLLayer).colorspace = windowColorSpace.cgColorSpace
```

Comment:
no forced casts. i still dislike the else case here.

Reply:
Need some help here. How else can I set color space on the `view.layer`? It's `CALayer`, right? Doesn't have `colorspace` property.

Also what's wrong with else, you mean the curly bracket?

Reply:
typecasting is okay, but you should not force typecasting by any means, eg as? instead of as!. the latter case will crash when failing. see https://docs.swift.org/swift-book/LanguageGuide/TypeCasting.html

the else is unnecessary there.
```
colorSpcace = defaultColorSpace
if someCondtion == true {
    colorSpcace = differentColorSpace
} 
```

Reply:
Unfortunately this way typecasting doesn't work. With as? I receive a compilation error about unwrapping CAOpenGLLayer, probably `view.layer` has a wrong type definition?

Reply:
something like this should haves worked. though i would need to see the specific error.
```
guard let newlayer = layer as? CAOpenGLLayer else {
      return
 }
// access newlayer here
```